src/phpDocumentor/Descriptor/Builder/PhpParser/ClassAssembler.php

```diff
@@ -31,22 +31,18 @@ class ClassAssembler extends AssemblerAbstract
      */
     public function create($data)
     {
-        $classDescriptor = new ClassDescriptor();
-
-        $this->assembleDocBlock($data->docBlock, $classDescriptor);
-
-        $classDescriptor->setFullyQualifiedStructuralElementName('\\' . $data->namespacedName->toString());
-        $classDescriptor->setName($data->name);
-        $classDescriptor->setLine($data->getLine());
-        $classDescriptor->setParent($data->extends ? '\\' . $data->extends->toString() : null);
-        $classDescriptor->setAbstract($data->isAbstract());
-        $classDescriptor->setFinal($data->isFinal());
-
-        $classDescriptor->setNamespace('\\' . $this->extractNamespace($data));
+        $classDescriptor = new ClassDescriptor(
+            new Fqsen('\\' . $data->namespacedName->toString()),
+            $data->docBlock,
+            $data->extends ? new ClassDescriptor(new Fqsen('\\' . $data->extends)) : null,
```

Comment:
This should just be the Fqsen of the superclass because it is only a relation between this class and its superclass; the actual class element for the superclass is built once it is encounted in the Files